The error (for the updated version) looks like
```
expected: (partial= {:a 1, :b []})
  actual: java.lang.AssertionError: Assert failed: partial= expects exactly 2 arguments
(clojure.core/= 2 1)
```

I've restructured this to be based on the arg count, and also to do the check inside the quote and not at macro expansion time, which produced cleaner error reports.

I tried to add tests for this but I can't find a permutation that actually captures the inner error. Any pointers for that?

--- a/test/metabase/test_runner/assert_exprs.clj
+++ b/test/metabase/test_runner/assert_exprs.clj
@@ -119,10 +119,9 @@
 
 (defmethod t/assert-expr 'partial=
   [message [_ expected actual :as form]]
-  `(do
-     (assert (= 2 ~(count (rest form))) "partial= expects exactly 2 arguments")
-     (t/do-report
-       (partial=-report ~message ~expected ~actual))))
+  (assert (= (count (rest form)) 2) "partial= expects exactly 2 arguments")
+  `(t/do-report
+    (partial=-report ~message ~expected ~actual)))
 
 (defn sql=-report
   [message expected query]

